client/rest-high-level/src/main/java/org/elasticsearch/client/security/user/privileges/GlobalPrivileges.java

```diff
@@ -37,64 +40,79 @@
+                    new ParseField(category));
+        }
     }
 
-    private final Set<? extends GlobalScopedPrivilege> applicationPrivileges;
+    private final Set<? extends GlobalOperationPrivilege> privileges;
 
     /**
-     * Constructs global privileges by bundling the set of application privileges.
+     * Constructs global privileges by bundling the set of privileges.
      * 
-     * @param applicationPrivileges
-     *            The privileges over applications.
+     * @param privileges
+     *            The privileges under a category and for an operation in that category.
      */
-    public GlobalPrivileges(Collection<? extends GlobalScopedPrivilege> applicationPrivileges) {
-        if (applicationPrivileges == null || applicationPrivileges.isEmpty()) {
-            throw new IllegalArgumentException("Application privileges cannot be empty or null");
+    public GlobalPrivileges(Collection<? extends GlobalOperationPrivilege> privileges) {
+        if (privileges == null || privileges.isEmpty()) {
+            throw new IllegalArgumentException("Privileges cannot be empty or null");
         }
-        this.applicationPrivileges = Collections.unmodifiableSet(new HashSet<>(Objects.requireNonNull(applicationPrivileges)));
-        final Set<String> allScopes = this.applicationPrivileges.stream().map(p -> p.getScope()).collect(Collectors.toSet());
-        if (allScopes.size() != this.applicationPrivileges.size()) {
-            throw new IllegalArgumentException(
-                    "Application privileges have the same scope but the privileges differ. Only one privilege for any one scope is allowed.");
+        // duplicates are just ignored
+        this.privileges = Collections.unmodifiableSet(new HashSet<>(Objects.requireNonNull(privileges)));
+        final Map<String, List<GlobalOperationPrivilege>> privilegesByCategoryMap =
+                this.privileges.stream().collect(Collectors.groupingBy(GlobalOperationPrivilege::getCategory));
+        for (final Map.Entry<String, List<GlobalOperationPrivilege>> privilegesByCategory : privilegesByCategoryMap.entrySet()) {
+            // all operations for a specific category
+            final Set<String> allOperations = privilegesByCategory.getValue().stream().map(p -> p.getOperation()).collect(Collectors.toSet());
+            if (allOperations.size() != privilegesByCategory.getValue().size()) {
+                throw new IllegalArgumentException("Different privileges for the same category and operation are not permitted");
+            }
         }
     }
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        final Map<String, List<GlobalOperationPrivilege>> privilegesByCategoryMap =
+                this.privileges.stream().collect(Collectors.groupingBy(GlobalOperationPrivilege::getCategory));
```

Comment:
Should we just keep this `Map` around?
We use it in the contructor, and in `toXContent`, couldn't it just be a field?

Reply:
I went with your suggestion. At first, I have resisted duplicating information, but since it's immutable and private, I suppose it's OK.